### DISCUSSION TITLE
ESP32 and 8port Relay - strange behavior

### DISCUSSION THREAD

**Discussion by chrstnkglr** [arendst/Tasmota#16089] [Q&A] [answered]
Hi all, 

I have a very strange behavior with an ESP32 running Tasmota. 
The ESP is connected to a 8 port relay.  

When I toggle port 1 of the relay with mqtt, it sometimes switches but not always. Also no mqtt message seems to arrive for this port. 
This is NOT the case with the other ports. Port 1 on the relay is GPIO PIN 13. 

What is also very strange is that when I toggle the relay port 7 for example, port 1 on the relay is switched as well. 

Do you have ideas what this can be related to and how I can find the error?

Thanks, Chris

**Comment by sfromis** [comment, marked as answer]
Watch the console with `Weblog 3` as it happens, to get details.

**Reply by chrstnkglr** [reply]
Yes, I turned on all debug levels and also did a ping to the ESP. 
I can toggle the Relay Port 1 to ON. As soon as I do a second toggle to OFF it crashes the ESP and there is no log available. Also ping switches to request timeout. ESP restarts. 
This also happens in 9 of 10 cases if I click in the web interface Relay Port 1

**Reply by sfromis** [reply]
Well, the details in the console are often key to understanding what happens

**Reply by barbudor** [reply]
I believe that sfromis mean "don't just enable the log levels but share with us"

While a software issue is still possible, the point that the ESP restarts/crashes make me think about an hardware problem.
How does it behave when the relay board is disconnected from the ESP?
Double check your cabling especially regarding power supplies.
Are you using separated power supplies for esp and relays?
If no, are you sure your PSU can handle both the ESP (500mA min) and relays (100mA per relay)
If yes, do you have a proper common ground with no ground loops?

**Comment by chrstnkglr** [comment]
I have researched further and found the root cause, but I need help on how to solve this problem. 
With the relay I switch the valves for my garden watering on ports 1 and 2. These valves are connected to a 24 VAC transformer (XBS BT-8). https://imgur.com/a/wDzqicz
The other ports of the relay switch normal 230V voltage.  
**As soon as I switch off the 24 VAC transformer I have no more problems and the ESP also switches port 1 of the relay absolutely error free.**
Do you have any ideas how I could solve the problem?

**Reply by barbudor** [reply]
Without a detailed and readable wiring diagram, I wouldn't even try to guess
